commands/common/typescript.go

```diff
@@ -0,0 +1,67 @@
+package common
+
+import (
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/jfrog/jfrog-cli-platform-services/model"
+)
+
+var (
+	tsNewInstancePattern   = regexp.MustCompile(`new\s+([A-Z][a-zA-Z0-9]+)\(`)
+	tsFieldTypePattern     = regexp.MustCompile(`[a-zA-Z0-9]+\s*:\s*([A-Z][a-zA-Z0-9]+)`)
+	tsFieldAccessPattern   = regexp.MustCompile(`\W([A-Z][a-zA-Z0-9]+)\.[a-zA-Z0-9]+`)
+	tsTypeInTypeParameters = regexp.MustCompile(`<([A-Z][a-zA-Z0-9]+)>`)
+	tsExcludeTypes         = []string{"PlatformContext"}
+	tsUnexportedType       = regexp.MustCompile(`^(class|type|interface|enum|const)\s+[A-Za-z_$][0-9A-Za-z_$]*`)
```

Comment:
Accept whitespace at the beginning
```suggestion
	tsUnexportedType       = regexp.MustCompile(`^\s*(class|type|interface|enum|const)\s+[A-Za-z_$][0-9A-Za-z_$]*`)
```